input/otlp/metadata_test.go

```diff
@@ -342,6 +344,74 @@ func TestResourceLabels(t *testing.T) {
 	}, modelpb.NumericLabels(metadata.NumericLabels))
 }
 
+func TestTranslateSpanValue(t *testing.T) {
+
+	m := pcommon.NewMap()
+	m.PutStr("s", "v")
+	m.PutBool("b", true)
+	m.PutInt("i", 1.0)
+	m.PutDouble("f", 1.0)
+
+	e := &modelpb.APMEvent{
+		Event:         &modelpb.Event{},
+		Span:          &modelpb.Span{},
+		Labels:        make(map[string]*modelpb.LabelValue),
+		NumericLabels: make(map[string]*modelpb.NumericLabelValue),
+	}
+
+	otlp.TranslateSpan(ptrace.SpanKindInternal, m, e)
+
+	assert.Equal(t, "v", e.GetLabels()["s"].Value)
+	assert.Equal(t, "true", e.GetLabels()["b"].Value)
+	assert.Equal(t, float64(1), e.GetNumericLabels()["i"].Value)
+	assert.Equal(t, float64(1), e.GetNumericLabels()["f"].Value)
+}
+
+func TestTranslateSpanSlice(t *testing.T) {
```

Comment:
If I look at this test without the context that this actually tests `setLabel`, I'd wonder why we test `TranslateSpan` only, but not `TranslateTransaction`. I think it makes sense to add a dimension to this table driven test so that it tests both `TranslateSpan` and `TranslateTransaction`. Span events also use `setLabel`, but `convertSpanEvent` is slightly complex, so I'm fine with leaving that out unless others feel strongly about it.